core/src/OpenMPTarget/Kokkos_OpenMPTarget_Parallel.hpp

```diff
@@ -1023,32 +1023,28 @@ struct ParallelReduceSpecialize<FunctorType, TeamPolicyInternal<PolicyArgs...>,
   // RangePolicy. Need a new implementation.
   static void execute_init_join(const FunctorType& f, const PolicyType& p,
                                 PointerType ptr, const bool ptr_on_device) {
-    const auto begin      = p.begin();
-    const auto end        = p.end();
     constexpr int HasInit = ReduceFunctorHasInit<FunctorType>::value;
 
-    const auto size = end - begin;
-
     const int league_size   = p.league_size();
     const int team_size     = p.team_size();
     const int vector_length = p.impl_vector_length();
 
+    auto begin = 0;
+    auto end   = league_size * team_size + team_size * vector_length;
+
     const size_t shmem_size_L0 = p.scratch_size(0, team_size);
     const size_t shmem_size_L1 = p.scratch_size(1, team_size);
 
     // FIXME_OPENMPTARGET: This would oversubscribe scratch memory since we are
     // already using the available scratch memory to create temporaries for each
     // thread.
-    if constexpr ((shmem_size_L0 + shmem_size_L1) > 0) {
+    if ((shmem_size_L0 + shmem_size_L1) > 0) {
       Kokkos::abort(
           "OpenMPTarget: Scratch memory is not supported in `parallel_reduce` "
           "over functors with init/join.");
     }
 
-    // Maximum active teams possible.
-    int max_active_teams = OpenMPTargetExec::MAX_ACTIVE_THREADS / team_size;
-    const auto nteams =
-        league_size < max_active_teams ? league_size : max_active_teams;
+    const auto nteams = league_size;
```

Comment:
Did you mean to change this?

Reply:
Yes the `nteams` should be equal to `league_size` . No need to include the `max_active_teams` in this case.